src/include/transaction/transaction_manager.h

```diff
@@ -22,79 +24,35 @@ class TransactionManager {
    * @param buffer_pool the buffer pool to use for transaction undo buffers
    * @param gc_enabled true if txns should be stored in a local queue to hand off to the GC, false otherwise
    */
-  explicit TransactionManager(common::ObjectPool<UndoBufferSegment> *buffer_pool, bool gc_enabled)
+  explicit TransactionManager(common::ObjectPool<storage::BufferSegment> *buffer_pool, bool gc_enabled)
```

Comment:
`explicit TransactionManager(common::ObjectPool<storage::BufferSegment> *const buffer_pool, const bool gc_enabled)`

Reply:
fine... if somehow that gives you satisfaction...